src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java

```diff
@@ -348,9 +327,121 @@ else if (output.contains("[error]")) {
     return result;
   }
 
-  private void showPantsMakeTaskMessage(String message, NotificationCategory type, Project project) {
-    NotificationData notification =
-      new NotificationData(PantsConstants.PANTS, message, type, NotificationSource.TASK_EXECUTION);
-    ExternalSystemNotificationManager.getInstance(project).showNotification(PantsConstants.SYSTEM_ID, notification);
+
+  private void showPantsMakeTaskMessage(String message, ConsoleViewContentType type, Project project) {
+    ConsoleView executionConsole = PantsConsoleManager.getOrMakeNewConsole(project);
+    // Create a filter that monitors console outputs, and turns them into a hyperlink if applicable.
+    Filter filter = new Filter() {
+      @Nullable
+      @Override
+      public Result applyFilter(String line, int entireLength) {
+        Optional<ParseResult> result = ParseResult.parseErrorLocation(line, ERROR_TAG);
+        if (result.isPresent()) {
+
+          OpenFileHyperlinkInfo linkInfo = new OpenFileHyperlinkInfo(
+            project,
+            result.get().getFile(),
+            result.get().lineNumber - 1, // line number needs to be 0 indexed
```

Comment:
1. corrected.
2. Manually tested by intentionally feeding it with line number -1, there is no crash, just the cursor is somewhere random.